presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PruneIndexJoinColumns.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.plan.IndexJoinNode;
+import io.prestosql.sql.planner.plan.PlanNode;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static io.prestosql.sql.planner.iterative.rule.Util.restrictChildOutputs;
+import static io.prestosql.sql.planner.plan.Patterns.indexJoin;
+
+public class PruneIndexJoinColumns
+        extends ProjectOffPushDownRule<IndexJoinNode>
+{
+    public PruneIndexJoinColumns()
+    {
+        super(indexJoin());
+    }
+
+    @Override
+    protected Optional<PlanNode> pushDownProjectOff(Context context, IndexJoinNode indexJoinNode, Set<Symbol> referencedOutputs)
+    {
+        ImmutableSet.Builder<Symbol> probeInputs = ImmutableSet.<Symbol>builder()
+                .addAll(referencedOutputs)
+                .addAll(Iterables.transform(indexJoinNode.getCriteria(), IndexJoinNode.EquiJoinClause::getProbe));
```

Comment:
Use Java streams instead of Guava Iterables. Since the streams APIs were added to Java, we've been slowly migrating away from Guava's implementation.